File: reo03-contatos/backend/src/main/java/br/com/contatos/contatos/services/ContatosService.java
package br.com.contatos.contatos.services;

import br.com.contatos.contatos.interfaces.IContatosService;
import br.com.contatos.contatos.models.Contatos;
import br.com.contatos.contatos.repository.ContatosRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.CrossOrigin;

import java.util.List;

@Service
@CrossOrigin(origins = "*")
public class ContatosService implements IContatosService {

    @Autowired
    ContatosRepository contatosRepository;

    private Contatos findContById(Long idContato){

        return contatosRepository.findFirstById(idContato);
    }

    @Override
    public List<Contatos> listar() {

        return contatosRepository.findAll();
    }

    @Override
    public Contatos cadastrar(Contatos contato) {

        return contatosRepository.save(contato);
    }

    @Override
    public Contatos editar(Contatos contato) {

        return contatosRepository.save(contato);
    }

    @Override
    public void deletar(Long id) {

        Contatos contato = contatosRepository.findFirstById(id);

        contatosRepository.delete(contato);
    }

    @Override
    public Contatos listarContatoUnico(Long idContato) {

        return findContById(idContato);
    }
}

File: reo04-contatos-cep/frontend/src/services/contatosService.js
import {ApiService} from './api.service'


const contatoService =  {

    listar: async() => {
        return ApiService.get('contatos/listar');
    },

    getContato: async(id) => {
        return ApiService.get(`contatos/${id}/buscar`);
    },

    cadastrar: async(contato) => {
        return ApiService.post(`contatos/cadastrar`, contato);
    },

    editar: async(contato, id) => {
        return ApiService.put(`contatos/${id}/editar`, contato);
    },

    deletar: async(id) => {
        return ApiService.delete(`contatos/${id}/deletar`);
    },

}

export default contatoService;